### ISSUE TITLE
Batch requests in `MLFlowLogger`

### ISSUE THREAD

**Issue by schmidt-jake** [Lightning-AI/lightning#15900]
## Proposed refactor

Update the mlflow API callsites in `MLFlowLogger` to use the more efficient [`MlflowcClient.log_batch`](https://github.com/mlflow/mlflow/blob/cce45c208630830b28a32642bca345d3d75749a1/mlflow/tracking/client.py#L908) API.

### Motivation

Improve logging performance and reduce the number of API calls.

### Pitch

`MLFlowLogger` currently executes individual API calls for every hyperparameter (in [`log_hyperparameters`](https://github.com/Lightning-AI/lightning/blob/a86584d6dd4d50388c7dcef4f3854b0e8355b346/src/pytorch_lightning/loggers/mlflow.py#L240)) and metric (in [`log_metrics`](https://github.com/Lightning-AI/lightning/blob/a86584d6dd4d50388c7dcef4f3854b0e8355b346/src/pytorch_lightning/loggers/mlflow.py#L253)) using `for` loops. By refactoring these two methods to use `log_batch`, we could reduce the number and overhead of MLFlowLogger API calls.

I'd be happy to open a PR for this.

cc @borda

**Comment by Borda** [comment]
@schmidt-jake are you interested in implementing this feature? :rabbit:

**Comment by schmidt-jake** [comment]
@Borda yep, happy to do it!
